### PR TITLE
Special SnowFood [Food Feature + Food-Related fixes]

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
🆑 Cobby
:add: All forms of food is now unique, meaning it can be renamed and its description edited!
:fix: replaces the food renaming with `unique_rename` [consistency!]
:tweak: Changes feedback of renaming / changing description to fit all items [like food!], not just weapons.
/🆑

Check CL for PR description.

Didn't think this needed to be atomized since it's all pertaining to food renaming [the last one is more or less], but I can separate the "food feature" if it's an issue.

### CODE REVIEW

**Comment by Armhulen** [comment]
speedmerge i need this now

**Comment by jughu** [comment]
This is hyena meatcake now made of 80% hyenas!

**Comment by Iamgoofball** [comment]
:+1:

**Comment by Cheridan** [comment]
tbh we shoulda just done this years ago instead of adding 500 different items that currently clog up 10 tabs of crafting menu

**Comment by Iamgoofball** [comment]
@Cheridan i can always go full goofkitchen again

**Comment by ExcessiveUseOfCobblestone** [comment]
honestly if you want to still do that you could probably give the cooks a special "Appealatomizer" device that reskins the objects to whatever food icon [certain ingredients in the custom food allow for different choices so you can still have bragging rights if you make a rare concoction] and just remove all those crafting options. It's not like you get to see a variety since the kitchen is highly reliant on botany, who's busy making meme plants [i'm guilty!].

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -19,11 +19,11 @@
 	..()
 
 /obj/item/weapon/proc/rename_weapon(mob/M)
-	var/input = stripped_input(M,"What do you want to name the weapon?", ,"", MAX_NAME_LEN)
+	var/input = stripped_input(M,"What do you want to name the [name]?", ,"", MAX_NAME_LEN)
 
 	if(!qdeleted(src) && M.canUseTopic(src, BE_CLOSE) && input != "")
 		name = input
-		M << "You name the weapon [input]. Say hello to your new friend."
+		M << "You have successfully changed the object's name to [input]."
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/weaponry.dm]: I don't like referring to it as an object.
store the old name and do:
```You have successfully renamed \the [oldname] to \the [newname]```

eg: "You have successfully renamed the spaghetti to Frank"

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -19,11 +19,11 @@
 	..()
 
 /obj/item/weapon/proc/rename_weapon(mob/M)
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/weaponry.dm]: bonus points:
get this off of ```/weapon``` and onto ```/item```, rename proc appropriately

Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/weapons/weaponry.dm]: hahahahahaha

**Pull request merged**